### DISCUSSION TITLE
Wazuh 4

### DISCUSSION THREAD

**Discussion by b4b857f6ee** [Security-Onion-Solutions/securityonion#4726] [Ideas] [answered]
Hello,

Any ETA to add Wazuh v4?
So we don't need to generate key to add agent :D

**Comment by lan94** [comment, marked as answer]
Wazuh 4.15 is already on their[ TODO list](https://github.com/Security-Onion-Solutions/securityonion/projects/24#card-64488590), but unfortunately it is moved since months always to the next update. At the moment it seems that Wazuh 4 will be implemented at 2.3.80. But you are not the only one who is interested in newest Wazuh version.

**Reply by b4b857f6ee** [reply]
Ok thanks.
It's for the fact, we don't need to add a key for each agent. So we can install it in automatic :)

**Reply by lan94** [reply]
You can already install Wazuh agents automatic, e.g. using [password registration](https://documentation.wazuh.com/3.13/user-manual/registering/password-authorization-registration.html) or [certificates](https://documentation.wazuh.com/3.13/user-manual/registering/host-verification-registration.html).

**Reply by b4b857f6ee** [reply]
Ok, thank you.
I'm working in Testing environment and trying to auto all the configuration, that's why the wazuh 4 is cool :).

**Comment by MadGeckoo** [comment]
Will there be support for wazuh kibana app in the 2.3.80? Or is there already some unofficial way how to get it to work? Thanks in advance :)

**Reply by lan94** [reply]
It is already possible to upgrade Wazuh yourself.  You could build new Wazuh Container with Security Onions base Wazuh Image, backup your Wazuh manager files, then insert and start new built container with connected backup files. But each upgrade will make problems when Security Onion is still using v3.13.

**Reply by b4b857f6ee** [reply]
Having the Wazuh interface like the Kibana inside SecurityOnion could be awesome...

**Reply by metapolitics** [reply]
Add the support the issue seeking out of the box installs for the wazuh kibana app

Related: 
#1748
#237
